src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -903,28 +771,28 @@ func (r *ClusterReconciler) createExternalNodesList(
 	return result, nil
 }
 
-func (r *ClusterReconciler) handleClusterDeletion(
+func (r *ClusterReconciler) removeFinalizers(
 	ctx context.Context, redpandaCluster *vectorizedv1alpha1.Cluster, l logr.Logger,
-) (reconcile.Result, error) {
+) error {
 	log := l.WithName("handleClusterDeletion")
-	log.V(logger.DebugLevel).Info("handling cluster deletion")
+	log.V(logger.DebugLevel).Info("handling finzalizers removal")
 	if controllerutil.ContainsFinalizer(redpandaCluster, FinalizerKey) {
 		log.V(logger.DebugLevel).Info("removing finalizers")
 		pods, err := r.podList(ctx, redpandaCluster)
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to list Pods: %w", err)
+			return fmt.Errorf("unable to list Pods: %w", err)
 		}
 		for i := range pods.Items {
 			if err := r.removePodFinalizer(ctx, &pods.Items[i], log); err != nil {
-				return ctrl.Result{}, fmt.Errorf(`unable to remove finalizer for pod "%s": %w`, pods.Items[i].GetName(), err)
+				return fmt.Errorf(`unable to remove finalizer for pod "%s": %w`, pods.Items[i].GetName(), err)
 			}
 		}
 		controllerutil.RemoveFinalizer(redpandaCluster, FinalizerKey)
 		if err := r.Update(ctx, redpandaCluster); err != nil {
```

Comment:
Just my two cents but I think that using a patch to remove a finalizer would be preferable here unless `r.Update` handles changes to `.ResourceVersion` automatically?

Reply:
I think Update does not changes `.ResourceVersion` automatically. I will change that.